internal/s3api/s3_gateway.go

```diff
@@ -16,6 +16,10 @@ import (
 	"github.com/wpnpeiris/nats-s3/internal/validation"
 )
 
+type S3GatewayOptions struct {
```

Comment:
This is a good abstraction.
I would include `natsServers string` and  `natsOptions []nats.Option` here as well.

Reply:
we can improve this in a separate PR.